tests/topotests/lib/evpn.py

```diff
@@ -0,0 +1,179 @@
+        else:
+            logger.info("Check Ping IPv6 from R1(vrf-101) to R2(vrf-101, fd01::2) OK")
+
+
+def evpn_print_nexthop_rmac(tgen, router):
+    output = tgen.gears[router].vtysh_cmd("show evpn next-hops vni all", isjson=False)
+    logger.info("==== result from {} show evpn next-hops vni all".format(router))
+    logger.info(output)
+    output = tgen.gears[router].vtysh_cmd("show evpn rmac vni all", isjson=False)
+    logger.info("==== result from {}: show evpn rmac vni all".format(router))
+    logger.info(output)
+
+
+def evpn_check_nexthop(
+    router,
+    vni=101,
+    ipv4="192.168.0.1",
+    ipv6="::ffff:192.168.0.1",
+    prefix_ipv4="10.0.101.1/32",
+    prefix_ipv6="fd01::1/128",
+    expected_paths=1,
+):
+    # Check IPv4
+    expected = {
+        "ip": ipv4,
+        "refCount": 1,
+        "prefixList": [{"prefix": prefix_ipv4, "pathCount": expected_paths}],
+    }
+    test_func = partial(
+        topotest.router_json_cmp,
+        router,
+        f"show evpn next-hops vni {vni} ip {ipv4} json",
+        expected,
+    )
+    _, result = topotest.run_and_expect(test_func, None, count=20, wait=1)
+    assert result is None, "evpn ipv4 next-hops check failed"
+
+    # Check IPv6
+    expected = {
+        "ip": ipv6,
+        "refCount": 1,
+        "prefixList": [{"prefix": prefix_ipv6, "pathCount": expected_paths}],
+    }
+    test_func = partial(
+        topotest.router_json_cmp,
+        router,
+        f"show evpn next-hops vni {vni} ip {ipv6} json",
+        expected,
+    )
+    _, result = topotest.run_and_expect(test_func, None, count=20, wait=1)
+    assert result is None, "evpn ipv6 next-hops check failed"
+
+
+def evpn_check_contexts(
+    router,
+    ipv4_only=False,
+    ipv6_only=False,
+    vni="101",
+    ipv4="192.168.0.2",
+    ipv6="::ffff:192.168.0.2",
+):
+    """
+    Check EVPN nexthops and RMAC number  are correctly configured
+    """
+    if ipv4_only:
+        expected = {
+            vni: {
+                "numNextHops": 1,
+                ipv4: {
+                    "nexthopIp": ipv4,
+                },
+            }
+        }
+    elif ipv6_only:
+        expected = {
+            vni: {
+                "numNextHops": 1,
+                ipv6: {
+                    "nexthopIp": ipv6,
+                },
+            }
+        }
+    else:
+        expected = {
+            vni: {
+                "numNextHops": 2,
+                ipv4: {
+                    "nexthopIp": ipv4,
+                },
+                ipv6: {
+                    "nexthopIp": ipv6,
+                },
+            }
+        }
+    result = topotest.router_json_cmp(
+        router, "show evpn next-hops vni all json", expected
+    )
+    assert result is None, "evpn next-hops check failed"
+
+    expected = {vni: {"numRmacs": 1}}
+    result = topotest.router_json_cmp(router, "show evpn rmac vni all json", expected)
+    assert result is None, f"evpn rmac number check failed"
```

Comment:
It looks the same as the RMAC check, does it make sense to duplicate it? (I know it has been for some time already but might be good to improve it)